File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Python/s863488345_comments_removed.py
import sys

def main():
    mapNum = int(sys.stdin.readline().strip())
    for i in range(mapNum):
        if i != 0:
            print()
        h, w = map(int, sys.stdin.readline().split())
        battleTown = []
        x = -1
        y = -1
        for j in range(h):
            line = sys.stdin.readline().strip()
            row = list(line)
            battleTown.append(row)
            for k in range(w):
                c = row[k]
                if c in ['^', 'v', '<', '>']:
                    x = j
                    y = k
        bomberActNum = int(sys.stdin.readline().strip())
        bomberAct = sys.stdin.readline().strip()
        current_x, current_y = x, y
        for action in bomberAct:
            if action == 'U':
                battleTown[current_x][current_y] = '^'
                if current_x - 1 >= 0 and battleTown[current_x-1][current_y] == '.':
                    battleTown[current_x-1][current_y] = '^'
                    battleTown[current_x][current_y] = '.'
                    current_x -= 1
            elif action == 'D':
                battleTown[current_x][current_y] = 'v'
                if current_x + 1 < h and battleTown[current_x+1][current_y] == '.':
                    battleTown[current_x+1][current_y] = 'v'
                    battleTown[current_x][current_y] = '.'
                    current_x += 1
            elif action == 'L':
                battleTown[current_x][current_y] = '<'
                if current_y - 1 >= 0 and battleTown[current_x][current_y-1] == '.':
                    battleTown[current_x][current_y-1] = '<'
                    battleTown[current_x][current_y] = '.'
                    current_y -= 1
            elif action == 'R':
                battleTown[current_x][current_y] = '>'
                if current_y + 1 < w and battleTown[current_x][current_y+1] == '.':
                    battleTown[current_x][current_y+1] = '>'
                    battleTown[current_x][current_y] = '.'
                    current_y += 1
            elif action == 'S':
                a = current_x
                b = current_y
                move_bomber(a, b, battleTown, h, w)
        for row in battleTown:
            print(''.join(row))

def move_bomber(a, b, battleTown, h, w):
    if not (0 <= a <= h and 0 <= b <= w):
        return
    direction = battleTown[a][b]
    if direction == '^':
        for i in range(a-1, -1, -1):
            if a <= 0:
                return
            if battleTown[a-1][b] == '*':
                battleTown[a-1][b] = '.'
                return
            elif battleTown[a-1][b] == '#':
                return
            else:
                a -= 1
                if a == 0:
                    return
    elif direction == 'v':
        for i in range(a+1, h):
            if battleTown[a+1][b] == '*':
                battleTown[a+1][b] = '.'
                return
            elif battleTown[a+1][b] == '#':
                return
            else:
                a += 1
                if a == h:
                    return
    elif direction == '<':
        for i in range(b-1, -1, -1):
            if battleTown[a][b-1] == '*':
                battleTown[a][b-1] = '.'
                return
            elif battleTown[a][b-1] == '#':
                return
            else:
                b -= 1
                if b == 0:
                    return
    elif direction == '>':
        for i in range(b+1, w):
            if battleTown[a][b+1] == '*':
                battleTown[a][b+1] = '.'
                return
            elif battleTown[a][b+1] == '#':
                return
            else:
                b += 1
                if b == w:
                    return

if __name__ == "__main__":
    main()  # End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s863488345_comments_removed.go
package main

import (
	"fmt"
)

func main() {
	var mapNum int
	fmt.Scan(&mapNum)
	for test := 0; test < mapNum; test++ {
		if test > 0 {
			fmt.Println()
		}
		var h, w int
		fmt.Scan(&h, &w)
		grid := make([][]byte, h)
		var x, y int
		for i := 0; i < h; i++ {
			var line string
			fmt.Scan(&line)
			grid[i] = []byte(line)
			for j := range grid[i] {
				c := grid[i][j]
				if c == '^' || c == 'v' || c == '<' || c == '>' {
					x, y = i, j
				}
			}
		}
		var bomberActNum int
		var bomberAct string
		fmt.Scan(&bomberActNum)
		fmt.Scan(&bomberAct)
		for _, action := range bomberAct {
			switch action {
			case 'U':
				if x > 0 && grid[x-1][y] == '.' {
					grid[x][y] = '.'
					x--
					grid[x][y] = '^'
				}
			case 'D':
				if x < h-1 && grid[x+1][y] == '.' {
					grid[x][y] = '.'
					x++
					grid[x][y] = 'v'
				}
			case 'L':
				if y > 0 && grid[x][y-1] == '.' {
					grid[x][y] = '.'
					y--
					grid[x][y] = '<'
				}
			case 'R':
				if y < w-1 && grid[x][y+1] == '.' {
					grid[x][y] = '.'
					y++
					grid[x][y] = '>'
				}
			case 'S':
				moveBomber(grid, x, y, h, w)
			}
		}
		for _, row := range grid {
			fmt.Println(string(row))
		}
	}
}

func moveBomber(grid [][]byte, a, b, h, w int) {
	dir := grid[a][b]
	switch dir {
	case '^':
		currentA := a
		for currentA > 0 {
			currentA--
			if grid[currentA][b] == '*' {
				grid[currentA][b] = '.'
				return
			} else if grid[currentA][b] == '#' {
				return
			}
		}
	case 'v':
		currentA := a
		for currentA < h-1 {
			currentA++
			if grid[currentA][b] == '*' {
				grid[currentA][b] = '.'
				return
			} else if grid[currentA][b] == '#' {
				return
			}
		}
	case '<':
		currentB := b
		for currentB > 0 {
			currentB--
			if grid[a][currentB] == '*' {
				grid[a][currentB] = '.'
				return
			} else if grid[a][currentB] == '#' {
				return
			}
		}
	case '>':
		currentB := b
		for currentB < w-1 {
			currentB++
			if grid[a][currentB] == '*' {
				grid[a][currentB] = '.'
				return
			} else if grid[a][currentB] == '#' {
				return
			}
		}
	}
}
// End of Code
